src/Console/Installer.php

```diff
@@ -0,0 +1,382 @@
+        static::$name = static::ask($io, 'What is your plugin\'s name?');
+        static::$vendor = $packagist = static::ask($io, 'What is your plugin\'s vendor name?');
+        static::$description = static::ask($io, 'What does your plugin do?');
+
+        while (!in_array(static::$configuration, ['Y', 'N'])) {
+            static::$configuration = static::ask($io, 'Does your plugin need configuration (Y/N)?', null, 'Y');
+        }
+
+        while (!in_array(static::$migrations, ['Y', 'N'])) {
+            static::$migrations = static::ask($io, 'Does your plugin need database migrations (Y/N)?', null, 'Y');
+        }
+
+        $githubUser = static::_gitConfig('github.user');
+        $dashedName = static::_dasherize(static::$name);
+
+        if (empty($packagist)) {
+            $packagist = static::ask(
+                $io,
+                'What is the Packagist\'s plugin vendor name?',
+                null,
+                $githubUser
+            );
+
+            static::$package = sprintf(
+                '%s/%s',
+                static::_dasherize($packagist),
+                $dashedName
+            );
+        }
+
+        static::$github = sprintf('%s/%s', $githubUser, 'cakephp-' . $dashedName);
+        static::$github = static::ask($io, 'What is the GitHub\'s plugin URL', null, static::$github);
+
+        static::$namespace = implode("\\", [static::$name, '']);
+        if (!empty(static::$vendor)) {
+            static::$namespace = static::$vendor . "\\" . static::$namespace;
+        }
+
+        static::$plugin = static::$name;
+        if (!empty(static::$vendor)) {
+            static::$plugin = static::$vendor . '/' . static::$name;
+        }
+
+        if (empty(static::$package)) {
+            static::$package = sprintf(
+                '%s/%s',
+                static::_dasherize(static::$vendor),
+                static::_dasherize(static::$name)
+            );
+        }
+
+        static::_customizeComposerFile();
+
+        $message = sprintf("<info>The %s plugin for Cake3 was successfully created.</info>\n");
+        $io->write($message, static::$plugin);
+    }
+
+    /**
+     * Post-install hook.
+
+     * Responsible of customizes all files, creates necessary directories
+     * before finally auto-destructing (007-style).
+     *
+     * @param \Composer\Script\Event $event Composer script event.
+     * @return void
+     */
+    public static function postInstall(Event $event = null)
+    {
+        $path = dirname(dirname(__DIR__)) . DIRECTORY_SEPARATOR;
+
+        static::_recursivelyReplacePlaceholders($path);
+
+        if (static::$configuration === 'Y' || static::$migrations === 'Y') {
+            mkdir($path . 'config');
+        }
+
+        if (static::$migrations === 'Y') {
+            mkdir($path . 'config' . DIRECTORY_SEPARATOR . 'Migrations');
+        }
+
+        unlink(__FILE__);
```

Comment:
Might be better to ask for confirmation before self destructing. It's quite like one might want to re-run the installer and we should account for that. Should also provide instructions how one can re-run the installers manually.